src/core/Akka/Event/EventStream.cs

```diff
@@ -29,8 +30,13 @@ public class EventStream : LoggingBus
     {
         private readonly bool _debug;
 
-        private readonly AtomicReference<Either<IImmutableSet<IActorRef>, IActorRef>> _initiallySubscribedOrUnsubscriber =
-            new AtomicReference<Either<IImmutableSet<IActorRef>, IActorRef>>();
+        // used to uniquely name unsubscribers instances, should there be more than one ActorSystem / EventStream
+        private static readonly AtomicCounter UnsubscribersCounter = new AtomicCounter(0);
+        private readonly AtomicReference<IActorRef> _unsubscriber = new AtomicReference<IActorRef>(ActorRefs.NoSender);
```

Comment:
Replaced with 1 unsubscriber `IActorRef` and a `ConcurrentSet<IActorRef>`, the latter of which records subscribers that have been started prior to the `ActorSystem` itself starting.